### PR TITLE
boost-build 1.78.0

### PR DESCRIPTION
**Pull request by SMillerDev**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by iMichka** [comment]
`/opt/homebrew/Cellar/boost-build/1.78.0/bin` and especially `/opt/homebrew/Cellar/boost-build/1.78.0/bin/b2` are not being installed, which makes the test fail. Weird.

**Comment by SMillerDev** [comment]
They don't even exist after the build

**Comment by iMichka** [comment]
Maybe the "known issues" and the patch is related? https://www.boost.org/users/history/version_1_78_0.html

File: Formula/boost-build.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class BoostBuild < Formula
   desc "C++ build system"
   homepage "https://www.boost.org/build/"
-  url "https://github.com/boostorg/build/archive/boost-1.77.0.tar.gz"
-  sha256 "17ad1addbc08d1cc6ef52f7140097915bc4904c28c7d6d733c4a1a20d40bbc1c"
+  url "https://boostorg.jfrog.io/artifactory/main/release/1.78.0/source/boost_1_78_0.tar.bz2"
```
Comment (Bo98) [inline Formula/boost-build.rb:4]: This is the full boost source - not the boost-build source.

If you really want to use the full Boost source, then you'll need to cd to `tools/build`.

Comment (iMichka) [reply Formula/boost-build.rb:4]: @SMillerDev can we go back to the GitHub archive?

**Review by iMichka — APPROVED** [review]
Here we go; fixed. Let's ship this.

**Pull request closed without merging**